src/lakefs_spec/spec.py

```diff
@@ -398,14 +407,25 @@ def ls(self, path: str | os.PathLike[str], detail: bool = True, **kwargs: Any) -
             if cache_entry is not None:
                 if not detail:
                     return [e["name"] for e in cache_entry]
-                return cache_entry
+                return cache_entry[:]
+
+        recursive = kwargs.get("recursive", False)
+        if "recursive" in kwargs:
+            del kwargs["recursive"]
```

Comment:
You could use `recursive = kwargs.pop("recursive", False)` here.

Reply:
Nice, didn't have that on my radar!